controllers/services/monitoring/monitoring_controller.go

```diff
@@ -0,0 +1,104 @@
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/updatestatus"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+const serviceName = "monitoring"
+
+// NewServiceReconciler creates a ServiceReconciler for the Monitoring API.
+func NewServiceReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ReconcilerFor(mgr, &servicesv1.Monitoring{}).
+		// operands - owned
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Secret{}).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&corev1.Service{}).
+		Owns(&corev1.PersistentVolumeClaim{}).
+		Owns(&monitoringv1.ServiceMonitor{}).
+		Owns(&monitoringv1.PrometheusRule{}).
+		// By default, a predicated for changed generation is added by the Owns()
+		// method, however for deployments, we also need to retrieve status info
+		// hence we need a dedicated predicate to react to replicas status change
+		Owns(&appsv1.Deployment{}, reconciler.WithPredicates(resources.NewDeploymentPredicate())).
+		// operands - openshift
+		Owns(&routev1.Route{}).
+		// operands - watched
+		//
+		// By default the Watches functions adds:
+		// - an event handler mapping to a cluster scope resource identified by the
+		//   components.opendatahub.io/part-of annotation
+		// - a predicate that check for generation change for Delete/Updates events
+		//   for to objects that have the label components.opendatahub.io/part-of
+		// or services.opendatahub.io/part-of set to the current owner
+		//
+		Watches(&extv1.CustomResourceDefinition{}).
+		// actions
+		WithAction(initialize).
+		WithAction(kustomize.NewAction(
+			kustomize.WithCache(),
+			// Those are the default labels added by the legacy deploy method
+			// and should be preserved as the original plugin were affecting
+			// deployment selectors that are immutable once created, so it won't
+			// be possible to actually amend the labels in a non-disruptive
+			// manner.
+			//
+			// Additional labels/annotations MUST be added by the deploy action
+			// so they would affect only objects metadata without side effects
+			// kustomize.WithLabel(labels.ODH.Component(componentName), "true"),
+			kustomize.WithLabel(labels.K8SCommon.PartOf, serviceName),
+		)).
+		WithAction(deploy.NewAction(
+			deploy.WithCache(),
+			deploy.WithFieldOwner(servicesv1.MonitoringInstanceName),
+			deploy.WithLabel(labels.PlatformPartOf, servicesv1.MonitoringServiceName),
+		)).
+		WithAction(updatestatus.NewAction(
+			updatestatus.WithSelectorLabel(labels.PlatformPartOf, servicesv1.MonitoringServiceName),
+		)).
+		WithAction(updateStatus).
```

Comment:
this name updateStatus, always get me confused, one for package one for function :D